tests/test_pfx.c

```diff
@@ -30,324 +18,404 @@
-
-        rec.prefix.ver = LRTR_IPV6;
-        rec.min_len = 128;
-        rec.max_len = 128;
-        ((uint64_t*) rec.prefix.u.addr6.addr)[1] = max_i;
-        ((uint64_t*) rec.prefix.u.addr6.addr)[0] = min_i + i;
-        assert(pfx_table_remove(&pfxt, &rec) == PFX_SUCCESS);
-    }
-    pfx_table_free(&pfxt);
-    printf("Done\n");
+	struct pfx_table pfxt;
+	struct pfx_record pfx;
+	enum pfxv_state res;
+	const uint32_t min_i = 0xFFFF0000;
+	const uint32_t max_i = 0xFFFFFFF0;
+	/* init table with huge number of records */
+	pfx_table_init(&pfxt, NULL);
+	printf("Inserting %u records\n", (max_i - min_i) * 3);
+	for (uint32_t i = max_i; i >= min_i; i--) {
+		pfx.min_len = 32;
+		pfx.max_len = 32;
+		pfx.socket = NULL;
+		pfx.asn = i;
+		pfx.prefix.u.addr4.addr = htonl(i);
+		pfx.prefix.ver = LRTR_IPV4;
+		assert(pfx_table_add(&pfxt, &pfx) == PFX_SUCCESS);
+		/* add same prefix, with diff asn */
+		pfx.asn = i + 1;
+		assert(pfx_table_add(&pfxt, &pfx) == PFX_SUCCESS);
+		/* add same prefix, with diff len */
+		pfx.min_len = 128;
+		pfx.max_len = 128;
+		pfx.prefix.ver = LRTR_IPV6;
+		((uint64_t *)pfx.prefix.u.addr6.addr)[1] = max_i;
+		((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
+		assert(pfx_table_add(&pfxt, &pfx) == PFX_SUCCESS);
+	}
+
+	/* verify validation of huge number of records */
+	printf("validating..\n");
+	for (uint32_t i = max_i; i >= min_i; i--) {
+		pfx.min_len = 32;
+		pfx.max_len = 32;
+		pfx.prefix.ver = LRTR_IPV4;
+		pfx.prefix.u.addr4.addr = htonl(i);
+		assert(pfx_table_validate(&pfxt, i,
+					  &pfx.prefix, pfx.min_len,
+					  &res) == PFX_SUCCESS);
+		assert(res == BGP_PFXV_STATE_VALID);
+		assert(pfx_table_validate(&pfxt, i + 1,
+					  &pfx.prefix, pfx.min_len,
+					  &res) == PFX_SUCCESS);
+		assert(res == BGP_PFXV_STATE_VALID);
+
+		pfx.min_len = 128;
+		pfx.max_len = 128;
+		pfx.prefix.ver = LRTR_IPV6;
+		((uint64_t *)pfx.prefix.u.addr6.addr)[1] = max_i;
+		((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
+
+		assert(pfx_table_validate(&pfxt, i + 1,
+					  &pfx.prefix, pfx.min_len,
+					  &res) == PFX_SUCCESS);
+		assert(res == BGP_PFXV_STATE_VALID);
+	}
+
+	/* verify removal of huge number of records */
+	printf("removing records\n");
+	for (uint32_t i = max_i; i >= min_i; i--) {
+		pfx.socket = NULL;
+		pfx.min_len = 32;
+		pfx.max_len = 32;
+		pfx.asn = i;
+		pfx.prefix.ver = LRTR_IPV4;
+		pfx.prefix.u.addr4.addr = htonl(i);
+		assert(pfx_table_remove(&pfxt, &pfx) == PFX_SUCCESS);
+
+		pfx.asn = i + 1;
+		assert(pfx_table_remove(&pfxt, &pfx) == PFX_SUCCESS);
+
+		pfx.prefix.ver = LRTR_IPV6;
+		pfx.min_len = 128;
+		pfx.max_len = 128;
+		((uint64_t *)pfx.prefix.u.addr6.addr)[1] = max_i;
+		((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
+		assert(pfx_table_remove(&pfxt, &pfx) == PFX_SUCCESS);
+	}
+
+	/* cleanup: free table */
+	pfx_table_free(&pfxt);
+	printf("mass_test successful\n");
 }
 
-int main()
+/**
+ * @brief Test basic ops on prefix table
```

Comment:
the brief line can be replaced with the following line.
The first line doesn't provide any information